src/common/GasService.sol

```diff
@@ -47,34 +47,34 @@ contract GasService is IGasService {
     constructor(uint128 maxBatchGasLimit_) {
         _maxBatchGasLimit = maxBatchGasLimit_;
 
-        // NOTE: The hardcoded values are taken from the EndToEnd tests. This should be automated in the future.
-        scheduleUpgrade = BASE_COST + 28514;
-        cancelUpgrade = BASE_COST + 8861;
-        recoverTokens = BASE_COST + 82906;
-        registerAsset = BASE_COST + 34329;
-        setPoolAdapters = BASE_COST + 398010; // using MAX_ADAPTER_COUNT
-        request = BASE_COST + 86084; // request deposit case
-        notifyPool = BASE_COST + 1154806; // create escrow case
-        notifyShareClass = BASE_COST + 1775916;
-        notifyPricePoolPerShare = BASE_COST + 30496;
-        notifyPricePoolPerAsset = BASE_COST + 35759;
-        notifyShareMetadata = BASE_COST + 13343;
-        updateShareHook = BASE_COST + 6415;
-        initiateTransferShares = BASE_COST + 89689;
-        executeTransferShares = BASE_COST + 70428;
-        updateRestriction = BASE_COST + 35992;
-        updateContract = BASE_COST + 53345;
-        requestCallback = BASE_COST + 186947; // approve deposit case
-        updateVaultDeployAndLink = BASE_COST + 2770342;
-        updateVaultLink = BASE_COST + 100567;
-        updateVaultUnlink = BASE_COST + 20814;
-        setRequestManager = BASE_COST + 30039;
-        updateBalanceSheetManager = BASE_COST + 35241;
-        updateHoldingAmount = BASE_COST + 220866;
-        updateShares = BASE_COST + 49968;
-        maxAssetPriceAge = BASE_COST + 27260;
-        maxSharePriceAge = BASE_COST + 26032;
-        updateGatewayManager = BASE_COST + 26287;
+        // NOTE: Below values should be updated using script/utils/benchmark.sh
+        scheduleUpgrade = BASE_COST + 66469;
+        cancelUpgrade = BASE_COST + 46876;
+        recoverTokens = BASE_COST + 123959;
+        registerAsset = BASE_COST + 76846;
+        setPoolAdapters = BASE_COST + 452173; // using MAX_ADAPTER_COUNT
+        request = BASE_COST + 186928;
+        notifyPool = BASE_COST + 1118719; // create escrow case
+        notifyShareClass = BASE_COST + 1818909;
+        notifyPricePoolPerShare = BASE_COST + 74705;
+        notifyPricePoolPerAsset = BASE_COST + 78425;
+        notifyShareMetadata = BASE_COST + 87934;
+        updateShareHook = BASE_COST + 64017;
+        initiateTransferShares = BASE_COST + 250624;
+        executeTransferShares = BASE_COST + 144865;
+        updateRestriction = BASE_COST + 81844;
+        updateContract = BASE_COST + 99339;
+        requestCallback = BASE_COST + 225405; // approve deposit case
+        updateVaultDeployAndLink = BASE_COST + 2820311;
+        updateVaultLink = BASE_COST + 152664;
+        updateVaultUnlink = BASE_COST + 101376;
+        setRequestManager = BASE_COST + 68197;
+        updateBalanceSheetManager = BASE_COST + 71761;
+        updateHoldingAmount = BASE_COST + 271358;
+        updateShares = BASE_COST + 151411;
```

Comment:
- High increment because of calling the snapshot (it seems before that was not taken into account)